source/config/config.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=paramgen.go Config
+
+package config
+
+import (
+	"strings"
+
+	"github.com/conduitio-labs/conduit-connector-redshift/common"
+)
+
+// Config contains source-specific configurable values.
+type Config struct {
+	common.Configuration
+	// OrderingColumn is a name of a column that the connector will use for ordering rows.
+	OrderingColumn string `json:"orderingColumn" validate:"required"`
+	// Snapshot is the configuration that determines whether the connector
+	// will take a snapshot of the entire table before starting cdc mode.
+	Snapshot bool `json:"snapshot" default:"true"`
+	// BatchSize is a size of rows batch.
+	BatchSize int `json:"batchSize" default:"1000"`
+}
+
+// Validate executes manual validations beyond what is defined in struct tags.
+func (c *Config) Validate() error {
+	// c.DSN has required validation handled in struct tag.
+
+	// c.Table required validation is handled in stuct tag
+	// handling "lowercase", "excludesall= " and "lte=127" validations.
+	if c.Table != strings.ToLower(c.Table) {
```

Comment:
these validations are duplicates of what we have in the destination configs, which as mentioned before can be moved to the commons config file, and we can call it here too

Reply:
Done.